src/utils.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Utility functions for the Hydra charm."""
+
+from typing import Dict
+from urllib.parse import urlparse, urlunparse
+
+
+def remove_none_values(dic: Dict) -> Dict:
+    """Remove all entries in a dict with `None` values."""
+    return {k: v for k, v in dic.items() if v is not None}
+
+
+def normalise_url(url: str) -> str:
```

Comment:
This function will be used by both Kratos and Hydra. But since it is a temporary hack until traefik solves this problem for us, I decided not to put it in a library and just duplicate the code.